quasar-repayer/src/repayment.js

```diff
@@ -0,0 +1,60 @@
+const numberToBN = require('number-to-bn');
+
+async function repay(rootChain, web3, quasar, account, token) {
+  // fetch exit queue for token
+  const exitQueue = await rootChain.getExitQueue(token);
+  console.log(exitQueue);
+  const blocknum = await web3.eth.getBlockNumber();
+  const timestampNow = (await web3.eth.getBlock(blocknum)).timestamp;
+  // fethc exits that are exitable from the queue
+  const exitableExitQueue = exitQueue.filter((exit) => exit.exitableAt < timestampNow);
+  const exitIdArr = exitableExitQueue.map((exit) => exit.exitId);
+
+  const peg = await rootChain.getPaymentExitGame();
+
+  // fetch all exitData
+  const exitsInQueue = await peg.contract.methods.standardExits(exitIdArr).call();
+  // exits of the quasar owner
+  const filteredExits = exitsInQueue.filter(
+    (exit) => exit.exitTarget.toLowerCase() === account.address.toLowerCase(),
+  );
+
+  if (filteredExits.length > 0) {
+    // get total amount to exit and repay
+    const totalAmount = filteredExits.reduce(
+      (acc, curr) => acc.add(numberToBN(curr.amount.toString())),
+      numberToBN(0),
+    );
+
+    const finalUtxoPos = filteredExits[filteredExits.length - 1].utxoPos;
+    // find index position of the last utxo of the quasar owner in the queue
+    const index = exitsInQueue.findIndex((exit) => exit.utxoPos === finalUtxoPos);
```

Comment:
It's not a big deal now, but I think in future we should put a limit on how many exits we automatically process.
As it is, we eagerly process all exits in the queue ahead of us, which could get expensive. Maybe if there are more than X exits ahead, we can wait a day or something to try save gas

Reply:
ah yes, makes sense. added an issue, as a reminder